Task №11

jacsmedvedev opened the pull request:
First commit of this task.

jacsmedvedev committed:
Task №11

First commit of this task.

strazhkoe commented on src/task_from_teacher_11/Main.java:
```diff
@@ -0,0 +1,94 @@
+			  .append("  ")
+			  .append((workers[i].getName()));	
+			System.out.println(sb.toString());
+		}
```
чтобы не копипастить эту логику, переопредели toString у рабочего

jacsmedvedev committed:
Task №11.

Исправил замечание - переопределил toString() в классе Worker.

strazhkoe commented on src/task_from_teacher_11/PaymentComparator.java:
```diff
@@ -0,0 +1,19 @@
+	public int compare(Worker w1, Worker w2 ) {
+		int result = Double.compare(w1.getPayment(), w2.getPayment());
+		if (result == 0) {
+			result = Double.compare(w1.getName().charAt(0), w2.getName().charAt(0)) * (-1);
```
Тут лучше сделать w1.getName().compareTo(w2.getName()). Тогда будет сравнение по всем буквам имени

strazhkoe commented on src/task_from_teacher_11/Worker.java:
```diff
@@ -0,0 +1,55 @@
+public abstract class Worker {
+
+	private String _name;
+	protected double _payment;
```
поле должно быть приватным

strazhkoe commented on src/task_from_teacher_11/Worker.java:
```diff
@@ -0,0 +1,55 @@
+		super();
+		_name = name;
+		_payment = payment;
+		_id = id;
```
Используй гет и сет методы

strazhkoe commented on src/task_from_teacher_11/Worker.java:
```diff
@@ -0,0 +1,55 @@
+		  .append(getPayment())
+		  .append("  ")
+		  .append(getName());	
+		return sb.toString();
```
тут можно делать склеивание через плюс. Все равно будет неявно создаваться билдер. Явное использование StringBuilder имеет смысл в более сложных сборах строки, например - через цикл

jacsmedvedev committed:
Task №11 commit#3

Исправил:
1.В class PaymentComparator сделал сравнение сравнение по всем буквам
имени;
2. protected double _payment на private double _payment;
3. Виесто _name = name;_payment = payment;_id = id включил set-methods;
4. в методе toString() заменил append на "+".

strazhkoe commented:
Добавь, пожалуйста, джавадоки на английском

jacsmedvedev committed:
Task №11 commit#4

Added javadocs and comments for public classes and methods in English.

strazhkoe commented on src/task_from_teacher_11/Main.java:
```diff
@@ -50,20 +32,22 @@ public static void main(String[] args) {
 		System.out.println("___________________________________");
 	}
 
+	/** method for printing workers and payments*/
```
все джавадоки подразумевают описание входящих параметров и возвращающих значений. Тут не описан параметр workers. Тут и ниже. Добавь пожалуйста

jacsmedvedev committed:
Task №11 commit#5

Corrected javadocs - added description of the parameters.

Akros2016 commented:
What's wrong ?

Akros2016 commented:
Жень, please, check this corrections.

strazhkoe commented on src/task_from_teacher_11/Worker.java:
```diff
@@ -46,6 +46,11 @@ public void setId(int id) {
+	 * 
+	 * @param x salary rate
+	 */
 	public abstract void avrgMonSalary(double x);
```
Метод должен что-то возвращать? Почему он публичный? Кто его должен вызывать и зачем если он void?

jacsmedvedev replied:
Публичный для того, чтобы использовать вычисление средней зп например для других категорий работников. Остальное исправил.

strazhkoe replied:
Попробовал изменить на protected - компилятор не ругался. Значит метод не используется за пределами иерархии классов.

strazhkoe commented on src/task_from_teacher_11/Main.java:
```diff
@@ -32,7 +32,10 @@ public static void main(String[] args) {
-	/** method for printing workers and payments*/
+	/**
+	 * Method for printing workers and payments.
+	 * @param workers
```
А описание к параметру?

jacsmedvedev committed:
Task №11 commit#6

- added descriptions of parameters to methods in Main class
- changed return type to method avrgMonSalary() in Worker class
from void to double
